### PR TITLE
Refactor v4 am

### PR DESCRIPTION
**Pull request by acollette**

### CODE REVIEW

File: src/asset-modules/UniswapV4/UniswapV4AM.sol [hunk 1]
```diff
@@ -12,11 +12,12 @@ import { FixedPoint128 } from "../../../lib/v4-periphery-fork/lib/v4-core/src/li
 import { FullMath } from "../../../lib/v4-periphery-fork/lib/v4-core/src/libraries/FullMath.sol";
 import { Hooks } from "./libraries/Hooks.sol";
 import { IPositionManager } from "./interfaces/IPositionManager.sol";
-import { IStateView } from "./interfaces/IStateView.sol";
+import { IPoolManager } from "../../../lib/v4-periphery-fork/lib/v4-core/src/interfaces/IPoolManager.sol";
```
Comment (Thomas-Smets) [inline src/asset-modules/UniswapV4/UniswapV4AM.sol:15]: Nitpicking, but should now be before IPos..

File: src/asset-modules/UniswapV4/UniswapV4AM.sol [hunk 2]
```diff
@@ -531,9 +510,6 @@ contract UniswapV4AM is DerivedAM {
     {
         // Also checks that msg.sender == Registry.
         super.processDirectWithdrawal(creditor, asset, assetId, amount);
-
```
Comment (Thomas-Smets) [inline src/asset-modules/UniswapV4/UniswapV4AM.sol:534]: Do we now still need to declare those functions?
we just call the one from AbstractDerivedRegistry now no?

File: src/asset-modules/UniswapV4/UniswapV4AM.sol [hunk 3]
```diff
@@ -240,13 +236,6 @@ contract UniswapV4AM is DerivedAM {
         // Calculate amount0 and amount1 of the accumulated fees.
         (uint256 fee0, uint256 fee1) = _getFeeAmounts(assetId, poolKey.toId(), info, liquidity);
 
-        // As the sole liquidity provider in a new pool,
```
Comment (Thomas-Smets) [inline src/asset-modules/UniswapV4/UniswapV4AM.sol:243]: Comment about lines 222 and 223, no need to cache token0 and token1.
Would immediately unwrap them in the _getKeyFromAsset

File: src/asset-modules/UniswapV4/UniswapV4AM.sol (hunk 2 above)
Comment (acollette) [reply src/asset-modules/UniswapV4/UniswapV4AM.sol:534]: think my mind was already in holiday for that PR..

**Pull request merged**